src/Sarif.ValidationTests/SarifValidatorTests.cs

```diff
@@ -113,6 +114,17 @@ private void ValidateDefaultDocument(IDictionary<Type, DefaultObjectPopulatingVi
             SarifLog clonedLog = sarifLog.DeepClone();
             clonedLog.ValueEquals(sarifLog).Should().BeTrue();
 
+            JObject jObjectSarifLog = JObject.Parse(toValidate);
+
+            // The following validations ensure NotYetAutoGenerated fields work as expected.
+
+            // see .src/sarif/NotYetAutoGenerated/ReportingConfiguration.cs (Rank property)
+            // see .src/sarif/NotYetAutoGenerated/Result.cs (Rank property)
+            bool isRankSerialized = RecursivePropertySearch(jObjectSarifLog, "rank");
+            isRankSerialized.Should().BeFalse();
+
+            // TODO: Add check for .src/sarif/NotYetAutoGenerated/Result.cs (resultKind and failureLevel properties)
+
```

Comment:
this test is needed, our current tests wont catch a change, lets me chat with you about this offline.